src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -13,44 +13,166 @@
 package frc.robot.subsystems;
 
 
-import frc.robot.commands.*;
-import edu.wpi.first.wpilibj.livewindow.LiveWindow;
+import edu.wpi.first.wpilibj2.command.CommandBase;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
-
-import edu.wpi.first.wpilibj.motorcontrol.MotorController;
-import edu.wpi.first.util.sendable.Sendable;
-import edu.wpi.first.wpilibj.Encoder;
 import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
 import frc.robot.Constants;
 
 public class IntakeSubsystem extends SubsystemBase {
     private final CANSparkMax intakeMotor;
+    private final RelativeEncoder intakeEncoder;
+    private final CANSparkMax midtakeMotor;
+    private final RelativeEncoder midtakeEncoder;
+
+    // Boolean for the way that the intake runs. True means forward, false means backwards:
+    private boolean runsForward;
+
+    // This motor raises and lowers the intake:
+    private final CANSparkMax raiseIntakeMotor;
+    private final RelativeEncoder raiseIntakeEncoder;
+
+    private double intakePosition; // This variable refers to the incline of the intake IN DEGREES
+
     public IntakeSubsystem() {
         intakeMotor = new CANSparkMax(Constants.kIntakeMotorPort, MotorType.kBrushless);
-        addChild("IntakeMotor", (Sendable) intakeMotor);
         intakeMotor.setInverted(false);
+        intakeEncoder = intakeMotor.getEncoder();
+        midtakeMotor = new CANSparkMax(Constants.kMidtakeMotorPort, MotorType.kBrushless); 
+        midtakeMotor.setInverted(false);
+        midtakeEncoder = midtakeMotor.getEncoder();
+
+        // raiseIntakeMotor:
+        raiseIntakeMotor = new CANSparkMax(Constants.kRaiseIntakeMotorPort, MotorType.kBrushless);
+        raiseIntakeMotor.setInverted(false);
+        raiseIntakeEncoder = raiseIntakeMotor.getEncoder();
+        raiseIntakeEncoder.setPosition(0);
+        intakePosition = 0;
+
+        // Intake starts off going forward:
+        runsForward = true;
     }
 
     @Override
     public void periodic() {
         // This method will be called once per scheduler run
+        double inVelocity = intakeEncoder.getVelocity();
+        double InCurrent = intakeMotor.getOutputCurrent(); 
+        SmartDashboard.putNumber("Intake Motor Speed (RPM)", inVelocity);
+        SmartDashboard.putNumber("Main Battery Current (A)", InCurrent);
```

Comment:
Bad copy/paste,
Shouldn't this be Intake Current?

Reply:
agreed, the string should be changed